### PR TITLE
mgr/dashboard: update QoS values when editing a Pool/RBD image

### PR DESCRIPTION
**Pull request by avanthakkar**
Fixes: https://tracker.ceph.com/issues/47900

Signed-off-by: Avan Thakkar <athakkar@redhat.com>

## Checklist
- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by avanthakkar** [comment]
Currently with the changes made, if the page is refreshed the value gets reset again to default values (i.e. 0 B/s or 0 IOPS), because every time the pool-form page is loaded the configuration form is created, so the value doesn't isn't retained. So any suggestion on how to retain the Qos field values even if the code is re-compiled or the page is refreshed? I tried replacing ngOnInit with ngOnChanges to detect changes calculated in getDirtyValues() (I tried storing values in both result object and config.value), but no luck!

**Comment by avanthakkar** [comment]
The issue is resolved now. In order to check with real case values, I still have to add e2e test cases.

**Comment by avanthakkar** [comment]
jenkins test make check

**Comment by avanthakkar** [comment]
jenkins test dashboard

**Comment by s0nea** [comment]
jenkins test dashboard

**Comment by avanthakkar** [comment]
jenkins test docs

**Review by epuertat — APPROVED** [review]
Great job! This was a tricky one @avanthakkar !

**Review by alfonsomthd — CHANGES_REQUESTED** [review]

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-configuration-form/rbd-configuration-form.component.spec.ts [hunk 1]
```diff
@@ -204,7 +205,7 @@ describe('RbdConfigurationFormComponent', () => {
     let data: any;
 
     beforeEach(() => {
-      component.initializeData = new EventEmitter<any>();
+      component.initializeData = new ReplaySubject<any>();
```
Comment (alfonsomthd) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-configuration-form/rbd-configuration-form.component.spec.ts:208]: If only the last value is to be replayed, better to set bufferSize to 1.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-configuration-form/rbd-configuration-form.component.spec.ts [hunk 2]
```diff
@@ -66,7 +67,7 @@ describe('RbdConfigurationFormComponent', () => {
 
   describe('test loading of initial data for editing', () => {
     beforeEach(() => {
-      component.initializeData = new EventEmitter<any>();
+      component.initializeData = new ReplaySubject<any>();
```
Comment (alfonsomthd) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-configuration-form/rbd-configuration-form.component.spec.ts:70]: If only the last value is to be replayed, better to set bufferSize to 1.

**Review by alfonsomthd — APPROVED** [review]
LGTM

**Comment by avanthakkar** [comment]
jenkins test dashboard

**Pull request merged**